public/app/plugins/panel/xychart/v2/SeriesEditor.tsx

```diff
@@ -119,174 +120,176 @@ export const SeriesEditor = ({
         </>
       )}
 
-      <Fragment key={formKey}>
-        <Field label="Frame">
-          <Select
-            placeholder={mapping === SeriesMapping.Auto ? 'All frames' : 'Select frame'}
-            isClearable={true}
-            options={context.data.map((frame, index) => ({
-              value: index,
-              label: `${getFrameDisplayName(frame, index)} (index: ${index}, rows: ${frame.length})`,
-            }))}
-            value={series.frame?.matcher.options}
-            onChange={(opt) => {
-              if (opt == null) {
-                delete series.frame;
-              } else {
-                series.frame = {
-                  matcher: {
-                    id: FrameMatcherID.byIndex,
-                    options: Number(opt.value),
-                  },
-                };
-              }
+      {selectedIdx >= 0 && series != null && (
```

Comment:
this is the main line preventing the crash right? the rest is just indentation noise? 😬

Reply:
yes, L35 and L123 are fixing the issue. The rest, indentation noise.. 😬